src/main/java/com/gamingmesh/jobs/Jobs.java

```diff
@@ -997,6 +998,17 @@ public static void action(JobsPlayer jPlayer, ActionInfo info, Block block, Enti
 
 	    Boost boost = getPlayerManager().getFinalBonus(jPlayer, noneJob);
 
+        JobsPrePaymentEvent JobsPrePaymentEvent = new JobsPrePaymentEvent(jPlayer.getPlayer(), noneJob, income, pointAmount);
+        Bukkit.getServer().getPluginManager().callEvent(JobsPrePaymentEvent);
+        // If event is canceled, don't do anything
+        if (JobsPrePaymentEvent.isCancelled()) {
+            income = 0D;
+            pointAmount = 0D;
+        } else {
+            income = JobsPrePaymentEvent.getAmount();
+            pointAmount = JobsPrePaymentEvent.getPoints();
+        }
```

Comment:
Remove this else

Reply:
Just the else or the else and what's in it?

Reply:
Fully, that contains in else variable.

Reply:
What's the point of the event if the income/pointAmount modifications from the event aren't used?

Reply:
There is no need to specify variations outside the class because if you register a new class and find the variations that you want, then create it.
An example:
```
private String s;

public somethingClass(String s) {
  this.s = s;
}
```
This is the class a variations registration. And the `s` will be available.